hack/istio/install-istio-via-sail.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+
+set -e
+
+if ! command -v yq &> /dev/null; then
+  echo "yq is required but it is either not installed or not in the PATH."
+  exit 1
+fi
+
+CUSTOM_INSTALL_SETTINGS=""
+PATCH_FILE=""
+
+# process command line args
+while [[ $# -gt 0 ]]; do
+  key="$1"
+  case $key in
+    -pf|--patch-file)
+      PATCH_FILE="$2"
+      shift;shift
+      ;;
+    -s|--set)
+      if [ -n "${CUSTOM_INSTALL_SETTINGS}" ]; then
+        CUSTOM_INSTALL_SETTINGS="${CUSTOM_INSTALL_SETTINGS} | $2"
+      else
+        CUSTOM_INSTALL_SETTINGS="$2"
+      fi
+      shift;shift
+      ;;
+    -h|--help)
+      cat <<HELPMSG
+Valid command line arguments:
+  -pf|--patch-file <name=value>:
+       filepath to a yaml file of an 'Istio' resource that will overlay the default 'Istio' resource. 
+       --patch-file /path/to/patch-file.yaml
+  -s|--set <name=value>:
+       Sets a name/value pair for a custom install setting. Some examples you may want to use:
+       --set installPackagePath=/git/clone/istio.io/installer
+  -h|--help:
+       this message
+HELPMSG
+      exit 1
+      ;;
+    *)
+      echo "ERROR: Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+helm upgrade sail-operator sail-operator \
+  --install \
+  --create-namespace \
+  --namespace sail-operator \
+  --repo https://istio-ecosystem.github.io/sail-operator
+
+K8S_GATEWAY_API_VERSION=$(curl --head --silent "https://github.com/kubernetes-sigs/gateway-api/releases/latest" | grep "location: " | awk '{print $2}' | sed "s/.*tag\///g" | cat -v | sed "s/\^M//g")
+echo "Installing Gateway API version ${K8S_GATEWAY_API_VERSION}"
+kubectl apply -k "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref=${K8S_GATEWAY_API_VERSION}"
+
+ISTIO_YAML=$(
+cat <<EOF
+apiVersion: sailoperator.io/v1alpha1
+kind: Istio
+metadata:
+  name: default
+spec:
+  version: v1.23.0
+  namespace: istio-system
+  updateStrategy:
+    type: InPlace
```

Comment:
I would also consider making this configurable. I know in the past I wanted to switch between InPlace and RevisionBased just to test the different behavior.

Reply:
Actually so that's the point of these two options:
```
  -pf|--patch-file <name=value>:
       filepath to a yaml file of an 'Istio' resource that will overlay the default 'Istio' resource. 
       --patch-file /path/to/patch-file.yaml
  -s|--set <name=value>:
       Sets a name/value pair for a custom install setting. Some examples you may want to use:
       --set 
```

The YAML in this file is just the default YAML if you don't apply any override but you can override any part of the YAML using either of these two options. With set you use it like this: `--set '.spec.version="v1.22.4"'` to specify a different version. If you're specifying many options you can use the patch file like this:
```
    local patch_file
    patch_file=$(mktemp)
    cat <<EOF > "$patch_file"
spec:
  values:
    global:
      meshID: mesh-default
      network: network-default
      multiCluster:
        clusterName: cluster-default
EOF
    "${SCRIPT_DIR}"/istio/install-istio-via-sail.sh --patch-file "$patch_file"
```

In both cases it just overrides the default YAML in this file and you can override any option you want.